core/src/main/java/org/opensearch/sql/ast/tree/Relation.java

```diff
@@ -45,54 +45,47 @@ public Relation(UnresolvedExpression tableName, String alias) {
   private String alias;
 
   /**
-   * Get original table name. Unwrap and get name if table name expression
-   * is actually an Alias.
-   * In case of federated queries we are assuming single table.
+   * Return table name.
    *
    * @return table name
    */
   public String getTableName() {
-    if (tableName.size() == 1 && ((QualifiedName) tableName.get(0)).first().isPresent()) {
-      return ((QualifiedName) tableName.get(0)).rest().toString();
-    }
-    return tableName.stream()
-        .map(UnresolvedExpression::toString)
-        .collect(Collectors.joining(COMMA));
+    return getTableQualifiedName().toString();
   }
 
   /**
-   * Get Catalog Name if present. Since in the initial phase we would be supporting single table
-   * federation queries, we are making an assumption of one table.
+   * Get original table name or its alias if present in Alias.
    *
-   * @return catalog name
+   * @return table name or its alias
    */
-  public String getCatalogName() {
-    if (tableName.size() == 1) {
-      if (tableName.get(0) instanceof QualifiedName) {
-        return ((QualifiedName) tableName.get(0)).first().orElse(null);
-      }
-    }
-    return null;
+  public String getTableNameOrAlias() {
+    return (alias == null) ? getTableName() : alias;
   }
 
   /**
-   * Return full qualified table name with catalog.
+   * Return alias.
    *
-   * @return fully qualified table name with catalog.
+   * @return alias.
    */
-  public String getFullyQualifiedTableNameWithCatalog() {
-    return tableName.stream()
-        .map(UnresolvedExpression::toString)
-        .collect(Collectors.joining(COMMA));
+  public String getAlias() {
+    return alias;
   }
 
   /**
-   * Get original table name or its alias if present in Alias.
+   * Get Qualified name preservs parts of the user given identifiers.
+   * This can later be utilized to determine Catalog,Schema and Table Name during
+   * Analyzer stage. So Passing QualifiedName directly to Analyzer Stage.
    *
-   * @return table name or its alias
+   * @return TableQualifiedName.
    */
-  public String getTableNameOrAlias() {
-    return (alias == null) ? getTableName() : alias;
+  public QualifiedName getTableQualifiedName() {
+    if (tableName.size() == 1) {
```

Comment:
why the if else logic is required? does it covered by stream.map.colloect()?

Reply:
Not it will loose the information of parts if I create with single name.
[/core/src/main/java/org/opensearch/sql/ast/expression/QualifiedName.java#L28-L41](https://github.com/vamsi-amazon/sql/blob/516071b8f69392a3e0190e502adb9872672fd5f4/core/src/main/java/org/opensearch/sql/ast/expression/QualifiedName.java#L28-L41)